src/cluegetter/http.go

```diff
@@ -458,13 +480,43 @@ func httpAbusersHandler(w http.ResponseWriter, r *http.Request) {
 
 	for rows.Next() {
 		result := &httpAbuserTop{}
-		rows.Scan(&result.Identifier, &result.Count)
+		rows.Scan(&result.SenderDomain, &result.SaslUsername, &result.Count)
 		data.SenderDomainTop = append(data.SenderDomainTop, result)
 	}
 
 	httpRenderOutput(w, r, "abusers.html", data, data.SenderDomainTop)
 }
 
+func httpGetSelectors(r *http.Request) (out []*httpAbuserSelector, err error) {
+	var selectors []*httpAbuserSelector
+
+	selectors = append(selectors, &httpAbuserSelector{
+		Name:     "sasl_username",
+		Text:     "Sasl Username",
+		Selected: r.FormValue("selector") == "sasl_username",
+	})
+
+	selectors = append(selectors, &httpAbuserSelector{
+		Name:     "sender_domain",
+		Text:     "Sender domain",
+		Selected: r.FormValue("selector") == "sender_domain",
+	})
+
+	hasSelectedSelector := false
+	for _, selector := range selectors {
+		if selector.Selected {
+			hasSelectedSelector = true
+			break
+		}
+	}
+
+	if !hasSelectedSelector && r.FormValue("selector") != "" {
+		return nil, errors.New("'" + r.FormValue("selector") + "' is not a valid value for parameter selector")
```

Comment:
Sorry for not noticing earlier. The conventino is to first display what's wrong, then show the specified value. E.g.:

Invalid selector specified: <selector>

Now you also get to add a capital at the start of the sentence #winrar :D